bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller_test.go

```diff
@@ -997,7 +998,12 @@ func TestBootstrapTokenRotationMachinePool(t *testing.T) {
 		if bytes.Equal(item.Data[bootstrapapi.BootstrapTokenExpirationKey], tokenExpires[0]) {
 			foundOld = true
 		} else {
-			g.Expect(string(item.Data[bootstrapapi.BootstrapTokenExpirationKey])).To(Equal(time.Now().UTC().Add(k.TokenTTL).Format(time.RFC3339)))
+			// To validate the expiration time is between just before the token rotate TTL and now TTL
+			expirationTime, err := time.Parse(time.RFC3339, string(item.Data[bootstrapapi.BootstrapTokenExpirationKey]))
+			g.Expect(err).NotTo(HaveOccurred())
+			timeJustBeforeRotateTTL := timeJustBeforeRotate.UTC().Add(k.TokenTTL).Truncate(time.Second)
+			timeNowTTL := time.Now().UTC().Add(k.TokenTTL).Truncate(time.Second)
+			g.Expect(expirationTime.Sub(timeJustBeforeRotateTTL) >= 0 && expirationTime.Sub(timeNowTTL) <= 0).To(BeTrue())
```

Comment:
Wondering if we can fix this using BeTemporally with some interval like e.g 10s or 30s

Reply:
can be done, lemme explore that.

Reply:
I feel following suites better than mentioning some random time, @fabriziopandini wdyt?

```golang
g.Expect(r.TimeFirst).Should(BeTemporally(">=", timeJustBeforeRotateTTL))
```

Reply:
@fabriziopandini added the approach in the recent commit, see if does make sense

Reply:
Even this is not suitable because the test can't really catch if TTL is added more than the default value, which may result in false-positive, may be I will stick to ~30s as mentioned.

Reply:
@fabriziopandini check the latest commit which uses BeTemporally with 10s as threshold